apps/condo/domains/common/components/LoginWithSBBOLButton.tsx

```diff
@@ -0,0 +1,50 @@
+// TODO: Restyle and refactor this button someday later
+import { useRouter } from 'next/router'
+import React from 'react'
+
+import { useIntl } from '@open-condo/next/intl'
+import { Button } from '@open-condo/ui'
+
+import { SberIconWithoutLabel } from './icons/SberIcon'
+
+import { useCheckTLSClientCert } from '../hooks/useCheckTLSClientCert'
+
+export const LoginWithSBBOLButton: React.FC<{ label?: string, block?: boolean, redirect?: string, checkTlsCert: boolean }> = ({
+    label,
+    block,
+    redirect,
+    checkTlsCert = true,
+}) => {
+    const intl = useIntl()
+    const LoginLabel = intl.formatMessage({ id: 'LoginBySBBOL' })
+    const router = useRouter()
+
+    const redirectToAuth = async () => {
+        const queryParams = redirect ? `?redirectUrl=${encodeURIComponent(redirect)}` : ''
+        const authUrl = `/api/sbbol/auth${queryParams}`
+        await router.push(authUrl)
+    }
+
+    const redirectToTlsPage = async () => {
+        await router.push('/tls')
+    }
+
+    const { loading, checkSSLClientCert } = useCheckTLSClientCert({
+        onSuccess: redirectToAuth,
+        onFail: redirectToTlsPage,
+    })
+
+    return (
+        <Button
+            key='submit'
+            type='secondary'
+            className='condo-btn-sbbol'
+            icon={<SberIconWithoutLabel/>}
+            onClick={() => checkTlsCert ? checkSSLClientCert() : redirectToAuth()}
```

Comment:
Maybe wrap this with `useCallback`?

Reply:
I see only extra complication for code with that.